packages/ckeditor5-engine/tests/view/observer/selectionobserver.js

```diff
@@ -167,26 +170,73 @@ describe( 'SelectionObserver', () => {
 			testUtils.sinon.stub( env, 'isGecko' ).value( true );
 		} );
 
-		it( 'should detect "restricted objects" in Firefox DOM ranges and prevent an error being thrown', done => {
-			const domRangeStubWithRestrictedObject = domDocument.createRange();
+		it( 'should detect "restricted objects" in Firefox DOM ranges and prevent an error being thrown', () => {
```

Comment:
Those tests are not testing SelectionObserver at all. You should test the mocked selection change, not the DomConverter directly.